### ISSUE TITLE
Should non-generic types be able to be called?

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#18681]
In the below program, `R()` is a call to the type constructor for `R`. Should this compile, given that `R` is not a generic type?

``` chapel
record R { var x: int; }

type t = R();
writeln(t:string);
```

(It does compile today).

**Comment by bradcray** [comment]
> Should this compile, given that R is not a generic type?

No, I'd be inclined to say that this should require being expressed as `type t = R;`

**Comment by mppf** [comment]
@bradcray - what about `int()` -- should that be a legal synonym for `int` aka `int(64)` ?

**Comment by bradcray** [comment]
> what about int() -- should that be a legal synonym for int aka int(64) ?

I'd be inclined to say "no".

> And, in my head, I think I have been imagining that this would work:
> 
> ```chapel
> type t = int;
> writeln(t(64):string);
> ```

My intuition is that that case shouldn't work either.  I think `int`—being a fully defaulted type—should resolve to `int(64)` barring any additional immediate information.  If/when we ever make good on the long-term idea you proposed years ago and that we've kicked around since then of making `int` more record-like (which I support, except that it's hard to prioritize), I'd prefer the user to have to write `type t = int(?);` to get the pattern above to work.

**Comment by BryantLam** [comment]
I skimmed the issue; is there a consideration for what a constrained-generics interface would have to look like for polymorphism? Records/POD types in the interface have to be called a particular way and (managed) classes would have to be called a different way. (I don't have a real question here. Just a comment.)
